src/components/application_manager/test/commands/mobile/set_app_icon_test.cc

```diff
@@ -144,6 +174,311 @@ TEST_F(SetAppIconRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
+
+  MockProtocolHandlerSettings mock_protocol_hendler_settings;
+  EXPECT_CALL(mock_protocol_hendler, get_settings())
+      .WillRepeatedly(ReturnRef(mock_protocol_hendler_settings));
+
+  const protocol_handler::MajorProtocolVersion max_supported_protocol_version =
+      protocol_handler::PROTOCOL_VERSION_3;
+  EXPECT_CALL(mock_protocol_hendler_settings, max_supported_protocol_version())
+      .WillRepeatedly(Return(max_supported_protocol_version));
+
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _)).WillOnce(Return(true));
+
+  request_->Run();
+}
+
+TEST_F(SetAppIconRequestTest,
+       CopyToIconStorage_MaximumStorageSizeLessFileSize_IconCopyingSkipped) {
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillRepeatedly(Return(mock_app_));
+
+  MockProtocolHandler mock_protocol_hendler;
+  EXPECT_CALL(app_mngr_, protocol_handler())
+      .WillRepeatedly(ReturnRef(mock_protocol_hendler));
+
+  MockProtocolHandlerSettings mock_protocol_hendler_settings;
+  EXPECT_CALL(mock_protocol_hendler, get_settings())
+      .WillRepeatedly(ReturnRef(mock_protocol_hendler_settings));
+
+  const protocol_handler::MajorProtocolVersion max_supported_protocol_version =
+      protocol_handler::PROTOCOL_VERSION_4;
+  EXPECT_CALL(mock_protocol_hendler_settings, max_supported_protocol_version())
+      .WillRepeatedly(Return(max_supported_protocol_version));
+
+  const uint32_t storage_max_size = 5u;
+  EXPECT_CALL(app_mngr_settings_, app_icons_folder_max_size())
+      .WillRepeatedly(ReturnRef(storage_max_size));
+
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _)).WillOnce(Return(true));
+
+  const std::vector<uint8_t> data(10, 10);
+  file_system::Write(kAppStorageFolder + "/" + kAppFolder + "/" + kSyncFileName,
+                     data);
+  request_->Run();
+}
+
+TEST_F(SetAppIconRequestTest, CopyToIconStorage_ZeroIcons_IconSavingSkipped) {
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_));
+
+  MockProtocolHandler mock_protocol_hendler;
+  EXPECT_CALL(app_mngr_, protocol_handler())
+      .WillRepeatedly(ReturnRef(mock_protocol_hendler));
+
+  MockProtocolHandlerSettings mock_protocol_hendler_settings;
+  EXPECT_CALL(mock_protocol_hendler, get_settings())
+      .WillRepeatedly(ReturnRef(mock_protocol_hendler_settings));
+
+  const protocol_handler::MajorProtocolVersion max_supported_protocol_version =
+      protocol_handler::PROTOCOL_VERSION_4;
+  EXPECT_CALL(mock_protocol_hendler_settings, max_supported_protocol_version())
+      .WillRepeatedly(Return(max_supported_protocol_version));
+
+  const uint32_t storage_max_size = 11u;
+  EXPECT_CALL(app_mngr_settings_, app_icons_folder_max_size())
+      .WillRepeatedly(ReturnRef(storage_max_size));
+
+  const uint32_t icons_amount = 0u;
+  EXPECT_CALL(app_mngr_settings_, app_icons_amount_to_remove())
+      .WillRepeatedly(ReturnRef(icons_amount));
+
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _)).WillOnce(Return(true));
```

Comment:
@BSolonenko I think you should check that `ManageMobileCommand` is called with `SUCCESS` result code in such cases

Reply:
Fixed on `ManageHMICommand(_)` in https://github.com/smartdevicelink/sdl_core/pull/1974/commits/647a7548e2badbf4476ca1700fccb7c85008103a